docs/sources/mimir/query/query-best-practices.md

```diff
@@ -0,0 +1,84 @@
+
+## Match your query time range to the evaluation frequency
+
+Align the time range of your query with how frequently the query is evaluated or how current the data needs to be. For instance, if a query runs every minute, such as for an alerting rule or a frequently refreshing dashboard panel, there's no need for it to look back for a long time period, such as 30 days.
+
+Queries with very long lookback periods can be computationally expensive. They require Mimir to process a large volume of historical data, even if only the latest value is of interest.
+
+## Filter early in the query
+
+Avoid running broad, large-scale queries and then filtering the results within Grafana. This approach is computationally expensive and inefficient. Instead, filter your data as early as possible in the query itself.
+
+Filtering data in the query provides the following benefits:
+
+- Reduced data transfer: Filtering at the query level means Mimir sends only the data you need to Grafana, which reduces network overhead.
+- Lower processing load: Mimir can apply filters at the storage layer, allowing it to discard irrelevant time series and data points before they are retrieved, processed, and aggregated. This significantly reduces the computational load on the Mimir cluster.
+- Improved query speed: Queries that are pre-filtered are generally faster to run and return results.
+
+Use these strategies to implement early filtering:
+
+- Use label selectors: Leverage precise label selectors within your PromQL query to narrow down the dataset from the start. For more information, refer to [Use Precise Label Selectors](#use-precise-label-selectors).
+- Apply functions and aggregations: If you need to transform or aggregate data, do so within the query. For example, `sum by (job) (metric_name)` is more efficient than retrieving all `metric_name` series and then summing them in Grafana.
+
+## Avoid high cardinality operations
+
+Avoid high cardinality operations to avoid performance issues or latency with your queries. High cardinality operations typically involve processing exceptionally large datasets.
+
+High cardinality typically results from the following circumstances:
+
+- Using a large number of labels
+- Attaching a large number of values to a label.
+
+To reduce cardinality in your queries, limit operations that involve many labels, such as the `group_by` function.
+
+## Use recording rules
+
+If you need to run a particularly expensive or complex query, consider creating a recording rule to minimize its load. Recording rules in Prometheus involve running queries at a predetermined time and then precomputing the query results to save for faster retrieval later.
+
+Recording rules are useful for the following types of queries:
+
+- Dashboard queries that run frequently
+- Complex aggregations across high-cardinality metrics
+- Queries spanning long time ranges
```

Comment:
These are both valid use cases for recording rules, but there's a trade-off here: if the recording rule results are only used infrequently, then they'll be paying the cost of evaluating and storing the recording rule results when they're not used. Perhaps we need to clarify this to emphasise that the best candidates for recording rules are queries that are both expensive and evaluated relatively frequently?